source/lambda/apiprocessor/lambda_function.py

```diff
@@ -1,31 +1,56 @@
- #  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance    #
- #  with the License. A copy of the License is located at                                                             #
- #                                                                                                                    #
- #      http://www.apache.org/licenses/LICENSE-2.0                                                                    #
- #                                                                                                                    #
- #  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES #
- #  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
- #  and limitations under the License.                                                                                #
- #####################################################################################################################
+#  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                           #
+#                                                                                                                    #
+#  Licensed under the Apache License, Version 2.0 (the License). You may not use this file except in compliance    #
+#  with the License. A copy of the License is located at                                                             #
+#                                                                                                                    #
+#      http://www.apache.org/licenses/LICENSE-2.0                                                                    #
+#                                                                                                                    #
+#  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES #
+#  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    #
+#  and limitations under the License.                                                                                #
+#####################################################################################################################
 
 import os
 import json
+import boto3
+from botocore.exceptions import ClientError
 
 from documents import getDocuments
 from document import getDocument, createDocument, deleteDocument
 from redact import redact
 from search import search, deleteESItem
 from kendraHelper import KendraHelper
 
+
+def redactHeadersFromLambdaEvent(lambdaEvent):
+    lambdaEvent.pop('headers', None)
+    lambdaEvent.pop('multiValueHeaders', None)
+    return lambdaEvent
+
+
+def validate_request(request):
```

Comment:
The request also have a type input passed we need to verify

Reply:
Also, NIT: can we rename this to validate_create_document_request since thats the only request we need to validate the input for

Reply:
updated and added more validation for different requests. Type looks like it is not used so i removed it.